lib/resource-providers/utils.ts

```diff
@@ -0,0 +1,27 @@
+import { IResource } from 'aws-cdk-lib/core';
+import { DummyProxy } from '../utils/proxy-utils';
+import { ResourceContext } from '../spi';
+import { v4 as uuidv4 } from 'uuid';
+
+export function getNamedResource<T extends IResource = IResource>(resourceName : string) : T {
+    return new Proxy({} as T, new DummyProxy((resourceContext: ResourceContext) => {
+        return resourceContext.get(resourceName) as T;
+    }));
+}
+
+
+export function getResource<T extends IResource = IResource>(fn: (context: ResourceContext) => T) : T {
```

Comment:
Could we add couple of lines of documentation of what this function is doing?

Reply:
addressed